server/datastore/mysql/queries.go

```diff
@@ -523,24 +527,51 @@ func (ds *Datastore) ListQueries(ctx context.Context, opt fleet.ListQueryOptions
 		}
 	}
 
-	if opt.MatchQuery != "" {
-		whereClauses += " AND q.name = ?"
-		args = append(args, opt.MatchQuery)
+	if opt.Platform != nil {
+		qs := fmt.Sprintf("%%%s%%", *opt.Platform)
+		args = append(args, qs)
+		whereClauses += ` AND q.platform LIKE ?`
```

Comment:
Empty `platform` means all platforms, so should this be:
```suggestion
		whereClauses += ` AND (q.platform LIKE ? OR q.platform = '')`
```

Reply:
Sounds good.

[This check](https://github.com/fleetdm/fleet/blob/749a2319a4909ddb9663e13df64a950c987e2d3a/server/service/queries.go#L130) also prevents passing in an empty string, at least at the service layer, right?

Reply:
Ah, I understand. Thanks!